src/frontend/src/lib/utils/nav.utils.ts

```diff
@@ -167,3 +170,31 @@ export const switchNetwork = async (networkId: Option<NetworkId>) => {
 
 	await goto(url, { replaceState: true, noScroll: true });
 };
+
+export const buildNftSearchUrl = ({
+	nft,
+	collection,
+	network
+}: {
+	nft?: Nft;
+	collection?: NftCollection;
+	network?: Network;
+}) => {
+	const url = new URL(`${window.location.origin}${AppPath.Nfts}`);
+
+	if (nonNullish(network?.id?.description)) {
+		url.searchParams.set(NETWORK_PARAM, network.id.description);
+	} else if (nonNullish(collection?.network?.id?.description)) {
+		url.searchParams.set(NETWORK_PARAM, collection.network.id.description);
+	}
```

Comment:
shouldn't the network be directly passed via nft param or via collection? for example

```suggestion
	if (nonNullish(nft?.network?.id?.description)) {
		url.searchParams.set(NETWORK_PARAM, network.id.description);
	} else if (nonNullish(collection?.network?.id?.description)) {
		url.searchParams.set(NETWORK_PARAM, collection.network.id.description);
	}
```

Reply:
updated it